packages/contracts/contracts/ChugSplashManagerProxy.sol

```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+
+import { Proxy } from "./libraries/Proxy.sol";
+import { ChugSplashRegistry } from "./ChugSplashRegistry.sol";
+import { IChugSplashManager } from "./interfaces/IChugSplashManager.sol";
+import { IChugSplashRegistry } from "./interfaces/IChugSplashRegistry.sol";
+
+/**
+ * @title ChugSplashManagerProxy
+ * @notice Designed to be upgradable only by the end user and to allow upgrades only to
+ *         new manager versions that whitelisted by the ChugSplashRegistry.
+ */
+contract ChugSplashManagerProxy is Proxy {
+    /**
+     * @notice Address of the ChugSplashRegistry.
+     */
+    ChugSplashRegistry public immutable registry;
+
+    /**
+     * @param _registry              The ChugSplashRegistry's address.
+     * @param _admin                 Owner of this contract.
+     * @param _initialImplementation The initial implementation of this contract. This will always
+     *                               be the V1 ChugSplashManager to ensure that manager proxy
+     *                               addresses are predictable.
+     */
+    constructor(
+        address _registry,
```

Comment:
nit: This can be `ChugSplashRegistry _registry` to reduce the number of type conversions